### PR TITLE
feat: rest framework enhancements

### PR DESCRIPTION
**Pull request by rahulguptajss**

### CODE REVIEW

File: cmd/collectors/rest/rest.go [hunk 1]
```diff
@@ -337,12 +342,26 @@ func (r *Rest) PollData() (*matrix.Matrix, error) {
...
-		for key, metric := range r.Matrix.GetMetrics() {
-			f := instanceData.Get(key)
+		for _, metric := range r.prop.metrics {
+			f := instanceData.Get(metric.name)
 			if f.Exists() {
-				if err = metric.SetValueFloat64(instance, f.Float()); err != nil {
-					r.Logger.Error().Err(err).Str("key", key).Str("metric", metric.GetName()).
-						Msg("Unable to set float key on metric")
+				if metr, ok := r.Matrix.GetMetrics()[metric.name]; !ok {
+					if metr, _ = r.Matrix.NewMetricFloat64(metric.name); err != nil {
```
Comment (cgrinds) [inline cmd/collectors/rest/rest.go:349]: would be slightly clearer to 1) create the metric by name and 2) update the map in this block and then remove the else and use it for both cases

File: cmd/collectors/rest/rest.go [hunk 2]
```diff
@@ -464,23 +483,23 @@ func (r *Rest) processEndPoints(data *matrix.Matrix) error {
 				}
 
 				for _, metric := range endpoint.prop.metrics {
-					f := instanceData.Get(metric)
+					f := instanceData.Get(metric.name)
 					if f.Exists() {
-						if metr, ok := data.GetMetrics()[metric]; !ok {
-							if metr, _ = data.NewMetricFloat64(metric); err != nil {
+						if metr, ok := data.GetMetrics()[metric.name]; !ok {
```
Comment (cgrinds) [inline cmd/collectors/rest/rest.go:488]: similar comment as above about minor refactoring

File: cmd/collectors/rest/rest.go (hunk 1 above)
Comment (Hardikl) [inline cmd/collectors/rest/rest.go:349]: err not used

Comment (rahulguptajss) [reply cmd/collectors/rest/rest.go]: done

File: cmd/collectors/rest/rest.go (hunk 2 above)
Comment (rahulguptajss) [reply cmd/collectors/rest/rest.go]: done

File: cmd/collectors/rest/rest.go (hunk 1 above)
Comment (rahulguptajss) [reply cmd/collectors/rest/rest.go]: done

File: cmd/collectors/rest/rest.go [hunk 3]
```diff
@@ -337,11 +342,20 @@ func (r *Rest) PollData() (*matrix.Matrix, error) {
...
+		for _, metric := range r.prop.metrics {
+			metr, ok := r.Matrix.GetMetrics()[metric.name]
+			if !ok {
+				if metr, err = r.Matrix.NewMetricFloat64(metric.name); err != nil {
+					r.Logger.Error().Err(err).
+						Str("name", metric.name).
+						Msg("NewMetricFloat64")
+				}
+			}
+			f := instanceData.Get(metric.name)
```
Comment (cgrinds) [inline cmd/collectors/rest/rest.go:354]: when err != nil is metr valid to use below?

File: cmd/collectors/rest/rest.go [hunk 4]
```diff
@@ -464,25 +478,20 @@ func (r *Rest) processEndPoints(data *matrix.Matrix) error {
...
-					f := instanceData.Get(metric)
+					metr, ok := data.GetMetrics()[metric.name]
+					if !ok {
+						if metr, err = data.NewMetricFloat64(metric.name); err != nil {
+							r.Logger.Error().Err(err).
+								Str("name", metric.name).
+								Msg("NewMetricFloat64")
+						}
+					}
+					f := instanceData.Get(metric.name)
```
Comment (cgrinds) [inline cmd/collectors/rest/rest.go:489]: when err != nil, is metr valid?

File: cmd/collectors/rest/templating.go [hunk 5]
```diff
@@ -28,16 +27,15 @@ func (r *Rest) initCache() error {
...
-		err                 error
 	)
 
-	if x := r.Params.GetChildContentS("object"); x != "" {
-		r.Object = x
-	} else {
-		r.Object = strings.ToLower(r.Object)
+	if r.Object == "" {
+		if x := r.Params.GetChildContentS("object"); x != "" {
+			r.Object = x
```
Comment (cgrinds) [inline cmd/collectors/rest/templating.go:34]: are we certain that x has already been lowercased?

Comment (rahulguptajss) [reply cmd/collectors/rest/templating.go:34]: we are lower casing it at line number 38 anyways

File: cmd/collectors/rest/rest.go (hunk 4 above)
Comment (rahulguptajss) [reply cmd/collectors/rest/rest.go:489]: metr is not added to matrix if there is an error. In current case, we check for duplicates.

**Pull request merged**